### DISCUSSION TITLE
Automate removal of ignored files

### DISCUSSION THREAD

**Discussion by LamanP** [gitextensions/gitextensions#10091] [Q&A] [answered]
### Feature description

Adding files to .gitignore and then remove them from the repo is tedious (see [this stackoverflow answer](https://stackoverflow.com/a/1274447/1111769)).

Please add a command that checks for any ignored files that are still in the index and removes them. After all that is what we want when we ignore files!

### Environment

- GitExtensions version: n/a
- GIT version: n/a
- OS version: n/a
- .NET version: n/a

**Comment by gerhardol** [comment, marked as answer]
There can be situations where you want to keep files in Git in ignored directories.

This could be added as a script.

**Reply by LamanP** [reply]
Of course. In that case don't use the command.

**Reply by pmiossec** [reply]
>Adding files to .gitignore and then remove them from the repo is tedious

Even with the "Stop tracking this file" we have in commit window and file tree contextual menus?

![image](https://user-images.githubusercontent.com/460196/181491796-428807a1-40df-4ef5-bbce-4137c86fb9c7.png)

>Please add a command that checks for any ignored files that are still in the index and removes them. 

The tricky thing is that GitExtensions can not implement a `.gitignore` parser so you will have to find the git command that list these files.
Maybe this link could help: https://stackoverflow.com/a/47351614/717372
ANd you could add it easily as a script in GitExtensions.
